src/components/show-edit-pod/show-edit-pod.test.js

```diff
@@ -0,0 +1,323 @@
+      {...props}
+    />
+  );
+};
+
+const UndoShowEditPodComponent = ({ ...props }) => {
+  return (
+    <ShowEditPod
+      title="With borders"
+      onEdit={() => {}}
+      onDelete={() => {}}
+      onUndo={() => {}}
+      saveText="Save"
+      deleteText="Delete"
+      cancelText="Cancel"
+      border
+      softDelete
+      editFields={<Textbox label="Field" labelInline labelAlign="right" />}
+      {...props}
+    />
+  );
+};
+
+context("Testing ShowEditPod component", () => {
+  describe("should render ShowEditPod component", () => {
+    it.each([
+      ["primary", "rgb(255, 255, 255)", "none"],
+      ["secondary", "rgb(242, 245, 246)", "none"],
+      ["tertiary", "rgb(230, 235, 237)", "none"],
+      ["tile", "rgb(255, 255, 255)", "rgba(2, 18, 36, 0.2) 0px 2px 3px 0px"],
+      ["transparent", "rgba(0, 0, 0, 0)", "none"],
+    ])(
+      "should check %s variant for ShowEditPod component",
+      (variant, color, boxShadow) => {
+        CypressMountWithProviders(<ShowEditPodComponent variant={variant} />);
+        showEditPod()
+          .should("have.css", "background-color", color)
+          .and("have.css", "box-shadow", boxShadow);
+      }
+    );
+
+    it.each([
+      [true, "1px solid rgb(204, 214, 219)"],
+      [false, "0px none rgb(0, 0, 0)"],
+    ])(
+      "should check when border is %s for ShowEditPod component",
+      (boolVal, border) => {
+        CypressMountWithProviders(<ShowEditPodComponent border={boolVal} />);
+        showEditPod().should("have.css", "border", border);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check children as %s for ShowEditPod component",
+      (children) => {
+        CypressMountWithProviders(
+          <ShowEditPodComponent>
+            {" "}
+            <Content title="Content title">{children}</Content>{" "}
+          </ShowEditPodComponent>
+        );
+        ContentShowEditPod().contains(children);
+      }
+    );
+
+    it.each(specialCharacters)(
+      "should check className as %s for ShowEditPod component",
+      (className) => {
+        CypressMountWithProviders(
+          <ShowEditPodComponent className={className} />
+        );
+        ClassShowEditPod().should("have.class", className);
+      }
+    );
+
+    it.each([
+      [true, "be.visible"],
+      [false, "not.exist"],
+    ])(
+      "should check when editing is %s for ShowEditPod component",
+      (boolVal, state) => {
+        CypressMountWithProviders(<ShowEditPodComponent editing={boolVal} />);
+        ContentFormShowEditPod().should(state);
+      }
+    );
+
+    it.each([
+      [false, "rgba(0, 0, 0, 0)"],
+      [true, "rgb(230, 235, 237)"],
+    ])(
+      "should check when softDelete is %s for ShowEditPod component",
+      (boolVal, backgroundColor) => {
+        CypressMountWithProviders(
+          <ShowEditPodComponent softDelete={boolVal} />
+        );
+        showEditPod().should("have.css", "background-color", backgroundColor);
```

Comment:
Part of the softDelete state includes the button in the pod being an 'Undo' button so you should assert that this is the case too.